### PR TITLE
Implementation of the states feature

### PR DESCRIPTION
**Pull request by Juliana-Dimeglio**
Added states, created a new loader and added functionality to de bottomnavigationBar buttons 

https://drive.google.com/file/d/1gh52JSyW9xJU-_paAxYxQGayTWABLOlZ/view?usp=sharing

### CODE REVIEW

File: lib/home_page.dart [hunk 1]
```diff
@@ -38,76 +36,107 @@ class _HomePageState extends State<HomePage> {
...
-              } else if (snapshot.hasError) {
-                return Text(
-                  snapshot.error.toString(),
-                );
-              }
-              return Center(
-                child: CircularProgressIndicator(),
+          builder: (context, snapshot) {
+            cantPages = ((snapshot.data?.response?.count ?? 0) /
+                    NumericConstants.cantPokemonXPage)
```
Comment (MarcosCGlobant) [inline lib/home_page.dart:49]: why the X on this constant?

File: lib/home_page.dart [hunk 2]
```diff
@@ -38,76 +36,107 @@ class _HomePageState extends State<HomePage> {
...
+                                setState(() {});
+                              }
+                            : () {},
+                        child: Icon(
+                          Icons.arrow_back,
+                          color: Colors.black,
+                        ),
+                      ),
+                      snapshot.data?.state == StateType.loading
+                          ? Text('')
```
Comment (MarcosCGlobant) [inline lib/home_page.dart:91]: this string can be a constant

File: lib/home_page.dart [hunk 3]
```diff
@@ -38,76 +36,107 @@ class _HomePageState extends State<HomePage> {
...
+                                  ),
+                                ),
+                                Text(
+                                  StringConstants.pageText +
+                                      ' $pageNumber - $cantPages',
+                                  style: TextStyle(
+                                    fontFamily: 'Pokemon',
+                                    fontSize: NumericConstants.pageTextFontSize,
+                                    color: secondPageColor,
+                                  ),
```
Comment (MarcosCGlobant) [inline lib/home_page.dart:114]: the TextStyles can be added into a styles file

File: lib/home_page.dart [hunk 4]
```diff
@@ -38,76 +36,107 @@ class _HomePageState extends State<HomePage> {
...
+                                _pokemonBloc.getPreviousPage();
+                                pageNumber--;
+                                setState(() {});
+                              }
+                            : () {},
+                        child: Icon(
+                          Icons.arrow_back,
+                          color: Colors.black,
+                        ),
+                      ),
```
Comment (FranciscoGaraventa) [inline lib/home_page.dart:89]: This button could be a widget, as parameter you can add it onPressed functionality

File: lib/home_page.dart [hunk 5]
```diff
@@ -38,76 +36,107 @@ class _HomePageState extends State<HomePage> {
...
+                                _pokemonBloc.getNextPage();
+                                pageNumber++;
+                                setState(() {});
+                              }
+                            : () {},
+                        child: Icon(
+                          Icons.arrow_forward,
+                          color: Colors.black,
+                        ),
+                      ),
```
Comment (FranciscoGaraventa) [inline lib/home_page.dart:133]: Here uses the same widget mentioned in the previous comment

File: lib/home_page.dart (hunk 1 above)
Comment (Juliana-Dimeglio) [reply lib/home_page.dart]: Fixed

File: lib/home_page.dart (hunk 2 above)
Comment (Juliana-Dimeglio) [reply lib/home_page.dart]: Fixed

File: lib/home_page.dart (hunk 3 above)
Comment (Juliana-Dimeglio) [reply lib/home_page.dart]: Fixed

File: lib/home_page.dart (hunk 4 above)
Comment (Juliana-Dimeglio) [reply lib/home_page.dart]: Fixed

File: lib/home_page.dart (hunk 5 above)
Comment (Juliana-Dimeglio) [reply lib/home_page.dart]: Fixed

**Review by MarcosCGlobant — APPROVED** [review]
LGTM

**Review by FranciscoGaraventa — APPROVED** [review]
LGTM!

**Pull request merged**